catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/RemoveCommandTest.java

```diff
@@ -13,119 +13,233 @@
+                .size(), equalTo(sizeOfFirstDeleteRequest));
+        assertThat(values.get(1)
+                .getAttributeValues()
+                .size(), equalTo(sizeOfSecondDeleteRequest));
+    }
 
+    @Test
+    public void testNullList() throws Exception {
         removeCommand.ids = null;
+        Object returnValue = removeCommand.executeWithSubject();
+        assertThat(returnValue, equalTo(-1));
+    }
 
+    @Test
+    public void testDeleteSingleMetacardFromStore() throws Exception {
+        configureMocks(1, 0);
+        removeCommand.ids = ImmutableSet.of(metacardList.get(0)
+                .getId());
         removeCommand.executeWithSubject();
+        verify(removeCommand.catalogFramework, times(1)).delete(isA(DeleteRequest.class));
+    }
 
-        assertThat(consoleOutput.getOutput(), containsString("Nothing to remove."));
+    @Test
+    public void testFilterMatchesNothing() throws Exception {
+        configureMocks(0);
+        Object returnValue = removeCommand.executeWithSubject();
+        assertThat(returnValue, equalTo(NO_FILTER_OR_IDS));
     }
 
-    private java.util.List<Metacard> getMetacardList(int amount) {
-        List<Metacard> metacards = new ArrayList<>();
+    @Test
+    public void testBothIdsAndFilter() throws Exception {
+        configureMocks(0);
+        removeCommand.ids = oneIdToDelete();
+        Object returnValue = removeCommand.executeWithSubject();
+        assertThat(returnValue, equalTo(SUCCESS));
+    }
 
-        for (int i = 0; i < amount; i++) {
-            String id = UUID.randomUUID()
-                    .toString();
-            MetacardImpl metacard = new MetacardImpl();
-            metacard.setId(id);
+    @Test
+    public void testCacheAndFilter() throws Exception {
+        removeCommand.cqlFilter = DUMMY_FILTER;
+        removeCommand.cache = true;
+        Object returnValue = removeCommand.executeWithSubject();
+        assertThat(returnValue, equalTo(CACHE_AND_FILTER_ILLEGAL));
+    }
 
-            metacards.add(metacard);
+    private void configureMocks(int... totalNumberOfMetacards) throws Exception {
+        List<List<Result>> queryResponseMockList = new ArrayList<>();
+        long hits = 0;
+        for (int numResultsForQuery : totalNumberOfMetacards) {
+            hits += numResultsForQuery;
+            List<Result> resultList = populateResultList(numResultsForQuery);
+            queryResponseMockList.add(resultList);
+
+            // add two copies of first result list as implementation of RemoveCommand
+            // calls queryResponse.getResults() three times per catalog framework query
+            // Calling getResults does NOT advance index of queryResponse
+            queryResponseMockList.add(resultList);
+            queryResponseMockList.add(resultList);
         }
+        when(queryResponse.getHits()).thenReturn(hits);
+        when(queryResponse.getResults()).thenReturn(queryResponseMockList.get(0),
```

Comment:
I think we're too closely tied to the implementation here. We should mock out the response from calling the `query` method on the catalog framework instead of the `getResults` on the actual mocked response. This will free us from being tied to the number of calls to `getResults`. See https://github.com/codice/ddf/commit/68f352ed2dd2d1554d6256977dbd6936023f452a (TestCswEndpoint) for an example